src/hazelcore/Statics_Pat.re

```diff
@@ -65,45 +104,56 @@ and syn_operand =
   | Inj(InHole(WrongLength, _), _, _)
   | TypeAnn(InHole(WrongLength, _), _, _) => None
   /* not in hole */
-  | Wild(NotInHole) => Some((Hole, ctx))
+  | Wild(NotInHole) => Some((Unknown(ModeSwitch), ctx))
   | Var(NotInHole, InVarHole(Free, _), _) => raise(UHPat.FreeVarInPat)
   | Var(NotInHole, InVarHole(ExpandingKeyword(_), _), _) =>
-    Some((Hole, ctx))
+    Some((Unknown(ModeSwitch), ctx))
   | Var(NotInHole, NotInVarHole, x) =>
-    Var.check_valid(
-      x,
-      Some((HTyp.Hole, Contexts.extend_gamma(ctx, (x, Hole)))),
-    )
+    switch (pattern_syn) {
+    | Moded => Some((Unknown(ModeSwitch), ctx))
+    | Unknown =>
+      let ty = HTyp.Unknown(ModeSwitch);
```

Comment:
everything just returns ModeSwitch in here now. I believe the correct behavior is that throughout the file, it should depend on the `pattern_syn` flag. When it is Moded then it should do Unknown(ModeSwitch) and if it is `Unknown` then it should do `Unknown(Internal)`?